File: Visual Basic .NET/Views3D/PresetItem.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Windows
Imports System.Windows.Media

Namespace Xceed.Wpf.DataGrid.Samples.Views3D
  Public Class PresetItem

    Private m_description As String

    Public Property Description() As String
      Get
        Return m_description
      End Get

      Set(ByVal value As String)
        m_description = value
      End Set
    End Property

    Private m_resourceDictionary As ResourceDictionary

    Public Property ResourceDictionary() As ResourceDictionary
      Get
        Return m_resourceDictionary
      End Get

      Set(ByVal value As ResourceDictionary)
        m_resourceDictionary = value
      End Set
    End Property

    Private m_preferredDataGridBackgroundBrush As Brush

    Public Property PreferredDataGridBackgroundBrush() As Brush
      Get
        Return m_preferredDataGridBackgroundBrush
      End Get

      Set(ByVal value As Brush)
        m_preferredDataGridBackgroundBrush = value
      End Set
    End Property

  End Class

End Namespace

File: CSharp/Views3D/PresetItem.cs
/*
 * Xceed DataGrid for WPF - SAMPLE CODE - Views 3D Sample Application
 * Copyright (c) 2007-2024 Xceed Software Inc.
 * 
 * [PresetItem.cs]
 *  
 * This class represents an item that will be used in the predefined
 * view settings combo box.
 * 
 * This file is part of the Xceed DataGrid for WPF product. The use 
 * and distribution of this Sample Code is subject to the terms 
 * and conditions referring to "Sample Code" that are specified in 
 * the XCEED SOFTWARE LICENSE AGREEMENT accompanying this product.
 *
 */

using System.Windows;
using System.Windows.Media;

namespace Xceed.Wpf.DataGrid.Samples.Views3D
{
  public class PresetItem
  {
    public string Description
    {
      get;
      set;
    }

    public ResourceDictionary ResourceDictionary
    {
      get;
      set;
    }

    public Brush PreferredDataGridBackgroundBrush
    {
      get;
      set;
    }
  }
}
